scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -531,10 +531,16 @@ object Execution {
    */
   private case class FlowDefExecution(result: (Config, Mode) => FlowDef) extends Execution[Unit] {
     protected def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) = {
-      lazy val future = cache
-        .runner
-        .validateAndRun(conf, mode)(result)
-        .map { m => ((), m) }
+      lazy val future = {
+        cache.writer match {
+          case ar: AsyncFlowDefRunner =>
+            ar.validateAndRun(conf, mode)(result).map { m => ((), Map(m)) }
+          case other =>
```

Comment:
I'm not sure we can easily do this differently with the current interface -- in that FlowDefExecution is tied to cascading. this match would eventually go away with a more cross platform interface I would think. Since we would want to eventually get references to FlowDef out of this file